lib/rspec/que/queue_up.rb

```diff
@@ -1,89 +1,155 @@
 require 'rspec/mocks/argument_list_matcher'
+require 'time'
 
 module RSpec
   module Que
     module Matchers
       class QueueUp
         def initialize(job_class = nil)
+          @matchers = [QueuedSomething.new]
+          @matchers << QueuedClass.new(job_class) if job_class
           @job_class = job_class
+
+          @stages = nil
         end
 
         def matches?(block)
           @before_jobs = enqueued_jobs.dup
           block.call
-          enqueued_something? && enqueued_correct_class? && with_correct_args?
+
+          @stages = []
+          @matched_jobs = new_jobs
+          @matchers.each do |matcher|
+            @stages << {matcher: matcher, candidates: @matched_jobs.dup}
+            @matched_jobs = @matched_jobs.delete_if {|job| !matcher.matches?(job) }
```

Comment:
`delete_if` works in place, so you can just say:
`@matched_jobs.delete_if { |job| !matcher.matches?(job) }`